### PR TITLE
[TestSDK] add VirtualNetworkPreparer, VnetNicPreparer

### PR DESCRIPTION
**Pull request by changlong-liu**
**Description**  

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S171

File: src/azure-cli-testsdk/azure/cli/testsdk/preparers.py [hunk 1]
```diff
@@ -257,6 +261,110 @@ def remove_resource(self, name, **kwargs):
...
+        self.cli_ctx = get_dummy_cli()
+        self.parameter_name = parameter_name
+        self.key = key
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.resource_group_key = resource_group_key
+        self.dev_setting_name = os.environ.get(dev_setting_name, None)
+
+    def create_resource(self, name, **kwargs):
+        if self.dev_setting_name:
+            return {self.parameter_name: self.dev_setting_name, }
```
Comment (MyronFanQiu) [inline src/azure-cli-testsdk/azure/cli/testsdk/preparers.py:287]:
```suggestion
            self.test_class_instance.kwargs[self.key] = name
            return {self.parameter_name: self.dev_setting_name, }
```

File: src/azure-cli-testsdk/azure/cli/testsdk/preparers.py [hunk 2]
```diff
@@ -257,6 +261,110 @@ def remove_resource(self, name, **kwargs):
...
+        self.cli_ctx = get_dummy_cli()
+        self.parameter_name = parameter_name
+        self.key = key
+        self.resource_group_parameter_name = resource_group_parameter_name
+        self.vnet_parameter_name = vnet_parameter_name
+        self.dev_setting_name = os.environ.get(dev_setting_name, None)
+
+    def create_resource(self, name, **kwargs):
+        if self.dev_setting_name:
+            return {self.parameter_name: self.dev_setting_name, }
```
Comment (MyronFanQiu) [inline src/azure-cli-testsdk/azure/cli/testsdk/preparers.py:337]:
```suggestion
            self.test_class_instance.kwargs[self.key] = name
            return {self.parameter_name: self.dev_setting_name, }
```

File: src/azure-cli-testsdk/azure/cli/testsdk/preparers.py [hunk 3]
```diff
@@ -257,6 +261,110 @@ def remove_resource(self, name, **kwargs):
...
 
+
+# pylint: disable=too-many-instance-attributes
+class VirtualNetworkPreparer(NoTrafficRecordingPreparer, SingleValueReplacer):
+    def __init__(self, name_prefix='clitest.vn',
+                 parameter_name='virtual_network',
+                 resource_group_parameter_name=None,
+                 resource_group_key=KEY_RESOURCE_GROUP,
+                 dev_setting_name='AZURE_CLI_TEST_DEV_VIRTUAL_NETWORK_NAME',
+                 random_name_length=24, key=KEY_VIRTUAL_NETWORK):
```
Comment (MyronFanQiu) [inline src/azure-cli-testsdk/azure/cli/testsdk/preparers.py:272]: Could we expose `sku` as well?

File: src/azure-cli-testsdk/azure/cli/testsdk/preparers.py (hunk 1 above)
Comment (changlong-liu) [reply src/azure-cli-testsdk/azure/cli/testsdk/preparers.py:288]: added here: https://github.com/Azure/azure-cli/pull/13871/commits/46bc336ff2b4f3443cf338c3d79debafa21f61b5

File: src/azure-cli-testsdk/azure/cli/testsdk/preparers.py (hunk 2 above)
Comment (changlong-liu) [reply src/azure-cli-testsdk/azure/cli/testsdk/preparers.py:339]: added here: https://github.com/Azure/azure-cli/pull/13871/commits/46bc336ff2b4f3443cf338c3d79debafa21f61b5

**Comment by yonzhan** [comment]
add to S172

**Pull request merged**